core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1948,10 +1957,22 @@ public String getUrlChildPrefix() {
         StaplerRequest req = Stapler.getCurrentRequest();
         if(req!=null)
             return getRootUrlFromRequest();
+        if (cachedRootUrl != null) {
+            return Util.ensureEndsWith(cachedRootUrl, "/");
```

Comment:
This violates contract of the method as it returns cached context path, not an absolute url.